src/client/grid/hex_grid.tsx

```diff
@@ -179,19 +189,34 @@ export function HexGrid({ onClick, onClickInterCity, rotation, fullMapVersion, h
     </div>}
     <div className={hexGridContainer}>
       <svg xmlns="http://www.w3.org/2000/svg"
-        fill="currentColor"
-        className={`bi bi-google ${hexGrid}`}
-        viewBox={`${viewBox.x} ${viewBox.y} ${viewBox.width} ${viewBox.height}`}
-        width={viewBox.width * zoom}
-        height={viewBox.height * zoom}
-        onClick={internalOnClick}>
-        {fullMapVersion && <DoubleHeightNumbers origin={internalViewBox} spacing={numberSpacing} rotation={rotation} grid={grid} size={size} />}
+           fill="currentColor"
+           className={`bi bi-google ${hexGrid}`}
+           viewBox={`${viewBox.x} ${viewBox.y} ${viewBox.width} ${viewBox.height}`}
+           width={viewBox.width * zoom}
+           height={viewBox.height * zoom}
+           onClick={internalOnClick}>
+        {fullMapVersion &&
+            <DoubleHeightNumbers origin={internalViewBox} spacing={numberSpacing} rotation={rotation} grid={grid}
+                                 size={size}/>}
+        <defs>
+          <filter id="cubeShadow" width={size/3} height={size/3}>
+            <feOffset in="SourceAlpha" dx={size/15} dy={size/15}/>
+            <feGaussianBlur stdDeviation={size/30}/>
+            <feBlend in="SourceGraphic" in2="blurOut"/>
+          </filter>
+        </defs>
         <g ref={ref}>
           {/* Rotating without a center moves it along the origin, but we rely on the viewBox calculation to make sure the view box fits the content. */}
           <Rotate rotation={rotation}>
             {mapSpaces}
-            {grid.connections.map((connection, index) => <InterCityConnectionRender key={index} highlighted={highlightedConnections?.some(c => interCityConnectionEquals(connection, c))} clickTargets={clickTargetsNormalized} onClick={onClickInterCity} size={size} connection={connection} />)}
-            {fullMapVersion && <SwedenProgressionGraphic />}
+            {riverLayer}
```

Comment:
Won't this render on top of track?

Reply:
Yup, you're right about that. I think there's some refactoring of `hexFactory(...)` so that different layers can be added incrementally instead of a single set of `mapSpaces`. I'm thinking of, `terrainSpaces`, `texturesLayer`, `trackSpaces`, `goodsSpaces`. The other nice thing about that refactor is that with goods as a separate layer, if a space has so many goods that it's overflowing, they won't get clipped by the edge of the hex.